remove memeber functionality

shreytheshreyas opened the pull request:
Implemented the remove member functionality for Team members

thatseant committed:
Implemented tests for TaskCommand

Includes tests for Task and TaskList classes.

TODO:
1) Abstract out System.out.println so print commands are easily testable.
2) Testing of error cases not done as errors not handled yet.

thatseant committed:
Merge branch 'master' into tasksTest

thatseant committed:
Modified tests to remove deadline check for tasks

thatseant committed:
Merge branch 'master' into tasksTest

shreytheshreyas committed:
Implemented the remove memeber functionality

shreytheshreyas committed:
Update DeleteTeamMemberCommand.java

thatseant committed:
Fixed Coding Style

thatseant committed:
Commented out failing test case

thatseant committed:
Commented out failing test case

thatseant commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -110,6 +111,11 @@ public static Command checkAction(String inputCommand) throws DukeExceptions {
             break;
+
+        case "remove":
+            memberName = inputCommand.split(" ")[1];
```
Should use the inputs array that Riaz implemented where he split the inputCommand into multiple words already.

thatseant replied:
Should use the inputs array that Riaz implemented where he split the inputCommand into multiple words already.

thatseant commented on src/main/java/seedu/duke/member/TeamMemberList.java:
```diff
@@ -18,12 +18,6 @@ public static void removeTeamMember(String name) {
         memberList.removeIf(member -> (member.name.contains(name)));
```
How do we handle deletion of members with duplicate names? One way could be not to allow duplicate members with same name to be created.

riazaham replied:
Hmm then should we implement this for projects and tasks also?

samuellleow replied:
I think that output-ing a warning for the user and ask them to provide a full name or have them provide a specific index would be better.

thatseant reviewed (commented):
So testing will be in a separate pull request after this.

riazaham committed:
Merge pull request #25 from thatseant/tasksTest

Implemented tests for TaskCommand

thatseant committed:
ExecuteCommand function returns string

String printed only in Duke.Java. This allows for better testing of ExecuteCommand and inner classes and methods.

thatseant committed:
Extract all System.out.println to Ui methods

New class Ui handles all output.

thatseant committed:
Fix Code Standards

riazaham committed:
Merge pull request #35 from thatseant/extractPrint

Extract Print to Ui Class

shreytheshreyas committed:
'Initial delete member functionality'

shreytheshreyas committed:
deleteMethod returns a string and include a function in memberlist class to check if a member is present in the list

shreytheshreyas committed:
replaced memberName with  with inputCommand array

shreytheshreyas committed:
'pull from intial PR'

shreytheshreyas committed:
Made program lines concise

riazaham reviewed (approved):
Looks good overall!

samuellleow reviewed (approved):
Great job on the delete member command.
